cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/datastore/DatastoreSchemaTests.java

```diff
@@ -405,4 +406,218 @@ public Integer call(SQLDatabase db) throws Exception {
+            assertAttachmentCount(datastore, "Document11",1092);
+            assertWinner(datastore, "Document11","7-rootecbaeb");
+
+            assertLeafCount(datastore, "Document12",540);
+            assertAttachmentCount(datastore, "Document12",1537);
+            assertWinner(datastore, "Document12","7-rootdcddcb");
+
+            assertLeafCount(datastore, "Document13",303);
+            assertAttachmentCount(datastore, "Document13",880);
+            assertWinner(datastore, "Document13","7-rootebbbbc");
+
+            assertLeafCount(datastore, "Document14",1);
+            assertAttachmentCount(datastore, "Document14",0);
+            assertWinner(datastore, "Document14","1-root");
+
+            assertLeafCount(datastore, "Document15",392);
+            assertAttachmentCount(datastore, "Document15",1127);
+            assertWinner(datastore, "Document15","7-rootcdddbd");
+
+            assertLeafCount(datastore, "Document16",1);
+            assertAttachmentCount(datastore, "Document16",0);
+            assertWinner(datastore, "Document16","1-root");
+
+            assertLeafCount(datastore, "Document17",326);
+            assertAttachmentCount(datastore, "Document17",1051);
+            assertWinner(datastore, "Document17","7-rootcddcbe");
+
+            assertLeafCount(datastore, "Document18",124);
+            assertAttachmentCount(datastore, "Document18",379);
+            assertWinner(datastore, "Document18","7-rootadecbd");
+
+            assertLeafCount(datastore, "Document19",1);
+            assertAttachmentCount(datastore, "Document19",0);
+            assertWinner(datastore, "Document19","1-root");
+
+            assertLeafCount(datastore, "Document20",248);
+            assertAttachmentCount(datastore, "Document20",724);
+            assertWinner(datastore, "Document20","7-rootbecbec");
+
+            assertLeafCount(datastore, "Document21",49);
+            assertAttachmentCount(datastore, "Document21",140);
+            assertWinner(datastore, "Document21","7-rootabaeac");
+
+            assertLeafCount(datastore, "Document22",1);
+            assertAttachmentCount(datastore, "Document22",0);
+            assertWinner(datastore, "Document22","1-root");
+
+            assertLeafCount(datastore, "Document23",258);
+            assertAttachmentCount(datastore, "Document23",805);
+            assertWinner(datastore, "Document23","7-rootbddabd");
+
+            assertLeafCount(datastore, "Document24",840);
+            assertAttachmentCount(datastore, "Document24",2358);
+            assertWinner(datastore, "Document24","7-rootedbebe");
+
+        } finally {
+            datastore.close();
+        }
+    }
+
+    // utility methods used by tests
+    private void assertLeafCount(Datastore datastore, String docId, int expectedLeafCount) {
+        int actualLeafCount = ((DatastoreImpl)datastore).getAllRevisionsOfDocument(docId).leafRevisions().size();
+        Assert.assertEquals(expectedLeafCount, actualLeafCount);
```

Comment:
Add a message, "Should get the expected number of leaves" so we get an obvious description of what has gone wrong if it fails:
>java.lang.AssertionError: Should get the expected number of leaves 
Expected :2357
Actual   :2310

is much easier to deal with than if there is no message.